File: Relation_IMS/Models/AuditLog.cs
using Finbuckle.MultiTenant;
using System.ComponentModel.DataAnnotations;

namespace Relation_IMS.Models
{
    public enum AuditType
    {
        Create = 1,
        Update = 2,
        Delete = 3
    }

    [MultiTenant]
    public class AuditLog
    {
        [Key]
        public int Id { get; set; }
        
        public int? UserId { get; set; }
        
        [Required]
        [MaxLength(20)]
        public string Type { get; set; } = null!;
        
        [Required]
        [MaxLength(100)]
        public string TableName { get; set; } = null!;
        
        [Required]
        public DateTime DateTime { get; set; }
        
        [MaxLength(255)]
        public string PrimaryKey { get; set; } = null!;
        
        public string? OldValues { get; set; }
        
        public string? NewValues { get; set; }
        
        public string? AffectedColumns { get; set; }
    }
}

File: Relation_IMS.React/src/types/auditLog.ts
export interface AuditLog {
    Id: number;
    UserId: number | null;
    UserName: string | null;
    UserRole: string | null;
    Type: 'Create' | 'Update' | 'Delete' | string;
    TableName: string;
    DateTime: string;
    PrimaryKey: string;
    OldValues: string | null;
    NewValues: string | null;
    AffectedColumns: string | null;
    Category: string;
    ActionLabel: string;
    Description: string;
    EntityId: string | null;
}

export interface AuditLogResponse {
    Records: AuditLog[];
    TotalCount: number;
    PageNumber: number;
    PageSize: number;
    TotalPages: number;
}

export interface AuditSummary {
    TotalCount: number;
    TodayCount: number;
    TrendPercentage: number;
    LastEntryTime: string | null;
}
